src/lib.rs

```diff
@@ -0,0 +1,148 @@
+use guest::prelude::*;
+use kubewarden_policy_sdk::wapc_guest as guest;
+
+use k8s_openapi::api::core::v1 as apicore;
+
+extern crate kubewarden_policy_sdk as kubewarden;
+use kubewarden::{protocol_version_guest, request::ValidationRequest, validate_settings};
+
+mod settings;
+use settings::Settings;
+
+#[no_mangle]
+pub extern "C" fn wapc_init() {
+    register_function("validate", validate);
+    register_function("validate_settings", validate_settings::<Settings>);
+    register_function("protocol_version", protocol_version_guest);
+}
+
+fn validate(payload: &[u8]) -> CallResult {
+    let validation_request: ValidationRequest<Settings> = ValidationRequest::new(payload)?;
+
+    let pvc = match serde_json::from_value::<apicore::PersistentVolumeClaim>(
+        validation_request.request.object,
+    ) {
+        Ok(pvc) => pvc,
+        Err(_) => {
+            // Not PVC, so we don't need to validate it
+            return kubewarden::accept_request();
+        }
+    };
+    if let (use_denied_class, Some(storage_class_name)) =
+        uses_denied_storage_class(&pvc, &validation_request.settings)
+    {
+        if use_denied_class && validation_request.settings.fallback_storage_class.is_some() {
+            return mutate_request(pvc, &validation_request.settings);
+        }
+        return kubewarden::reject_request(
+            Some(format!(
+                "storage class \"{}\" is not allowed",
+                storage_class_name
+            )),
+            None,
+            None,
+            None,
+        );
+    }
+    kubewarden::accept_request()
```

Comment:
I don't like to have too many nested levels of code. it makes the code hard to read. In this case we don't really need to do this `if let ... = `.

Please get rid of the `if let`, just do:

```rust
    let (use_denied_class, Some(storage_class_name)) = uses_denied_storage_class(&pvc, &validation_request.settings);
    if use_denied_class {
            if validation_request.settings.fallback_storage_class.is_some() {
               return mutate_request(pvc, &validation_request.settings);
            }
           return kubewarden::reject_request(
                Some(format!(
                  "storage class \"{}\" is not allowed",
                  storage_class_name
               )),
              None,
              None,
              None,
           );
        }     
    }
    kubewarden::accept_request()
```